### ISSUE TITLE
pyright complain "component" is not exported from module "solara"

### ISSUE THREAD

**Issue by zhuoqiang** [widgetti/solara#465]
pyright complains `"component" is not exported from module "solara"` for the following code, line 3

```python
import solara

@solara.component
class Page
    pass
```

turns out it is not a false alarm, see https://github.com/microsoft/pyright/issues/5929

> By default, all imports in a py.typed library are considered private unless they are explicitly re-exported. To indicate that an imported symbol is intended to be re-exported, the maintainers of this library would need to use one of the techniques documented [here](https://microsoft.github.io/pyright/#/typed-libraries?id=library-interface)

so I guess instead of 

```python
from reacton import component
```

it could be updated to re-exported using the redundant form to make pyright happy

```python
from reacton import component as component
```

**Comment by Lundez** [comment]
I think adding the following to `.vscode/settings.json` solved this for me 👍 
```json
    "python.analysis.packageIndexDepths": [
        {
            "name": "",
            "depth": 5
        }
    ],
```

**Comment by Jhsmit** [comment]
I have the same issue (with pylance)
Not only `solara.component` but also `solara.use_state` (basically any reacton import)

unfortunately @Lundez' fix doesnt seem to help

**Comment by edan-bainglass** [comment]
@Jhsmit true, an explicit `__all__` would help! For now, to calm `pylance`, I replace `solara` imports with more specific (deep) ones.
```python
from solara.core import component

@component
def MyComponent():
    ...
```
A bit more work, but acceptable at least in my use case. There are `TODO` tags regarding `__all__` in the `solara` module, so hopefully coming soon :)
